database/database.go

```diff
@@ -0,0 +1,115 @@
+package database
+
+import (
+	"dashboard/model"
+	"database/sql"
+
+	"golang.org/x/crypto/bcrypt"
+)
+
+var db *sql.DB
+
+//InitialiseDb to start db server
+func InitialiseDb() error {
+	var err error
+	db, err = sql.Open("mysql", "root:mysql1234@tcp(localhost:3306)/dashboard")
+	return err
+}
+
+//CloseDb to close DB connection
+func CloseDb() {
+	db.Close()
+}
+
+//GetUserByUsername func to get user by username
+func GetUserByUsername(username string) (*model.UserProfile, error) {
+	var err error
+	var profile *model.UserProfile
+	stmtOut, err := db.Prepare("SELECT * FROM user_profile WHERE username  = ? ")
+	if err != nil {
+		return nil, err
+	}
+	defer stmtOut.Close()
+	err = stmtOut.QueryRow(username).Scan(&profile.Username, &profile.FirstName, &profile.LastName, &profile.Password, &profile.Gender, &profile.Country, &profile.Age, &profile.Email)
+	return profile, err
+}
+
+//EncryptPassword to secure passwords
+func EncryptPassword(userpassword string) string {
+	hash, err := bcrypt.GenerateFromPassword([]byte(userpassword), 4)
+	if err != nil {
+		return err.Error()
+	}
+	userpassword = string(hash)
+	return userpassword
+}
+
+//InsertUserProfile func to insert user profile
+func InsertUserProfile(user *model.UserProfile) error {
+	var profile *model.UserProfile
+	var err error
+	profile, err = GetUserByUsername(user.Username)
+	if profile == nil && err != nil {
+		user.Password = EncryptPassword(user.Password)

```

Comment:
What if there is an error in EncryptPassword and its returning error string from line 41. Add check for error.